From bfbb38e18253ac1a237fcb45c8c053b57ede3f57 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Thu, 11 Nov 2021 13:23:04 -0800
Subject: [PATCH] egl/gallium/dri: Adding DRM_FORMAT_ABGR16161616 support

Adding DRM_FORMAT_ABGR16161616 support to egl/gallium/dri

v2: Fixed typo

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 src/egl/drivers/dri2/egl_dri2.c         | 1 +
 src/gallium/frontends/dri/dri_helpers.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/src/egl/drivers/dri2/egl_dri2.c b/src/egl/drivers/dri2/egl_dri2.c
index b1e4a9dfe13..abc8261a29b 100644
--- a/src/egl/drivers/dri2/egl_dri2.c
+++ b/src/egl/drivers/dri2/egl_dri2.c
@@ -2651,6 +2651,7 @@ dri2_num_fourcc_format_planes(EGLint format)
    case DRM_FORMAT_BGRA1010102:
    case DRM_FORMAT_XBGR16161616F:
    case DRM_FORMAT_ABGR16161616F:
+   case DRM_FORMAT_ABGR16161616:
    case DRM_FORMAT_YUYV:
    case DRM_FORMAT_YVYU:
    case DRM_FORMAT_UYVY:
diff --git a/src/gallium/frontends/dri/dri_helpers.c b/src/gallium/frontends/dri/dri_helpers.c
index a2c864ae101..418c9249b2c 100644
--- a/src/gallium/frontends/dri/dri_helpers.c
+++ b/src/gallium/frontends/dri/dri_helpers.c
@@ -436,6 +436,9 @@ static const struct dri2_format_mapping dri2_format_table[] = {
       { DRM_FORMAT_XBGR16161616F, __DRI_IMAGE_FORMAT_XBGR16161616F,
         __DRI_IMAGE_COMPONENTS_RGB,       PIPE_FORMAT_R16G16B16X16_FLOAT, 1,
         { { 0, 0, 0, __DRI_IMAGE_FORMAT_XBGR16161616F } } },
+      { DRM_FORMAT_ABGR16161616,  __DRI_IMAGE_FORMAT_ABGR16161616,
+        __DRI_IMAGE_COMPONENTS_RGBA,      PIPE_FORMAT_R16G16B16A16_UNORM, 1,
+        { { 0, 0, 0, __DRI_IMAGE_FORMAT_ABGR16161616 } } },
       { __DRI_IMAGE_FOURCC_RGBA16161616, __DRI_IMAGE_FORMAT_ABGR16161616,
         __DRI_IMAGE_COMPONENTS_RGBA,      PIPE_FORMAT_R16G16B16A16_UNORM, 1,
         { { 0, 0, 0, __DRI_IMAGE_FORMAT_ABGR16161616 } } },
-- 
GitLab

